ex6.sh

```diff
@@ -0,0 +1,19 @@
+
+# If no argument is provided, set default value to 50
+if [ -z "$1" ]; then
+  num_files=50
+else
+  num_files=$1
+fi
+
+# Create folder with current date
+folder_name="zero_$(date +%Y-%m-%d)_files"
+mkdir "$folder_name"
+
+# Create files with required format
+for (( i=1; i<=$num_files; i++ ))
+do
+  filename="File_${i}.dat"
+  printf '%0.s0' $(seq 1 $i) > "$folder_name/$filename"

```

Comment:
Notice that the demand was to save nullbytes (\x00) in the file and not ascii char 0 (\x30)